### PR TITLE
Merging latest development - just to keep things in  sync.  Should not impact any CI.

### PR DESCRIPTION
**Pull request by tclune**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [ ] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [ ] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: generic3g/OuterMetaComponent.F90 [hunk 1]
```diff
@@ -315,25 +337,115 @@ recursive subroutine initialize(this, importState, exportState, clock, unusable,
 
       integer :: status, userRC
       type(ChildComponent), pointer :: child
+      type(OuterMetaComponent), pointer :: child_meta
       type(ChildComponentMapIterator) :: iter
 
+      associate (phase => get_phase_index(this%phases_map%of(ESMF_METHOD_INITIALIZE), phase_name='GENERIC_INIT_GRID', rc=status))
+        if (status == _SUCCESS) then
+           call ESMF_GridCompInitialize(this%user_gridcomp, importState=importState, exportState=exportState, &
+                clock=clock, userRC=userRC, phase=phase, _RC)
```
Comment (atrayano) [inline generic3g/OuterMetaComponent.F90:346]: Could we have the userRC and _RC always to be the last two arguments of the ESMF calls

Comment (tclune) [reply generic3g/OuterMetaComponent.F90:346]: NP.  Please use the "add a suggestion" (+/- box) to make such requests in the future.  Makes it easier to manage.
```suggestion
                clock=clock, phase=phase, userRC=userRC, _RC)
```

File: generic3g/OuterMetaComponent.F90 [hunk 2]
```diff
@@ -315,25 +337,115 @@ recursive subroutine initialize(this, importState, exportState, clock, unusable,
...
+      integer, optional, intent(out) :: rc
 
+      integer :: status, userRC
+      type(ChildComponent), pointer :: child
+      type(ChildComponentMapIterator) :: iter
+
+      associate (phase => get_phase_index(this%phases_map%of(ESMF_METHOD_INITIALIZE), phase_name='DEFAULT', rc=status))
+        if (status == _SUCCESS) then
+           call ESMF_GridCompInitialize(this%user_gridcomp, importState=importState, exportState=exportState, &
+                clock=clock, userRC=userRC, phase=phase, _RC)
```
Comment (atrayano) [inline generic3g/OuterMetaComponent.F90:384]:
```suggestion
                clock=clock, phase=phase, userRC=userRC, _RC)
```

**Comment by tclune** [comment]
Looks like I have some debugging/workarounds to do.  Intel's failure is almost certainly a compiler bug and needs a workaround.   GFortran's is run time, so hard to know what it is stumbling on.

**Comment by tclune** [comment]
Pinging @mathomp4 to enjoy the latest commit (GFortran workaround)

**Comment by mathomp4** [comment]
> Pinging @mathomp4 to enjoy the latest commit (GFortran workaround)

Well that is ugly and crazy!

The CI still isn't happy with Intel. I wonder if we need to move to Intel 2021.7? Or if it needs a full on workaround?

**Comment by tclune** [comment]
It may just be that I have not pushed the Intel fix from my laptop.  I know I dealt with it ...

**Comment by mathomp4** [comment]
Obviously, I'll need to force this in with superpowers due to #1746. @tclune Are you done with your changes? If so, we can press the Update Branch button to get this into MAPL3.

That said, weirdly, the GNU GCM run failed in a different way than in #1746. Complaining about a bad pointer or something? I looked on discover and the GNU MAPL3 GCM run there failed like #1746 with DODCS.

**Comment by tclune** [comment]
Let me take one more stab this morning.   When I last looked yesterday things had not failed, so I was optimistic.

**Pull request merged**